### ISSUE TITLE
Unintended Removal Of End Sequence

### ISSUE THREAD

**Issue by skippyr** [skippyr/becho#6]
If `becho` is used with itself as a subcommand it can remove end sequences of already executed commands, which can lead to colors crossing command's boundaries. For example:

![wrong_end_sequence](https://user-images.githubusercontent.com/119078587/222579843-b3cd7eb2-507a-453a-8b03-10684bd541f7.png)

Issue happens in `v1.2.0`.

Output of previous command in verbose is:

```bash
Arguments {
    is_to_escape: false,
    is_bold: false,
    is_underline: false,
    is_crossed_out: false,
    is_italic: false,
    foreground_color: "normal",
    background_color: "normal",
    is_dimmed: false,
    separator: " ",
    case: "normal",
    is_no_end_new_line: false,
    number_of_repetitions: 1,
    is_to_ignore_empty: false,
    left_indentation: "",
    width: 80,
    is_verbose: true,
    text_fragments: [
        "\u{1b}[1m[\u{1b}[38;5;10mok]\u{1b}[0m more",
    ],
}
[ok] more
```

**Comment by skippyr** [comment]
A possible solution to this issue is to make the function `remove_end_sequences` only remove the last occurrence of a sub string instead of removing all it found when using the `replace` method.

**Comment by skippyr** [comment]
There was indeed a bug that was making styles going through sub command boundaries, which was fixed by commit d57fb4234363b2f27737450705b899c9a4306f55.

However, the command showed in this issue will look like it still having an "issue", but that is due to how `becho` is meant to work: `becho` apply styles per words. In the case of the command show previously, now the output, with the fix, is:

![issue](https://user-images.githubusercontent.com/119078587/222764695-0f53fcfb-fa31-4ff2-9aa5-ed0f9dd445a1.png)

As can be seen, `becho` applies the color green and puts an end sequence right after it, as it was supposed to work. The bold style is applied to the word `[ok]`, placing an end sequence right after `]`. Now, it is possible to see that between `ok` and `]` there is an intended end sequence that makes the character `]` reset its styles and do not have bold.

I think it will be better to apply styles per word than per characters as this can make debugging really a nightmare and wrapping, most of the times, will work perfectly fine with this algorithm.

But, this "issue" have a workaround, which is to separate `[ok]` and apply bold styles to each part instead of making `becho` consider it as one word:

![solution](https://user-images.githubusercontent.com/119078587/222766760-cb04ada6-95a7-42b2-bf64-945d49036c2a.png)

**Comment by skippyr** [comment]
Now, talking about the "elephant in the room" itself: there was a bug that was making end sequences of sub commands be removed. For example:
![bug](https://user-images.githubusercontent.com/119078587/222767671-ee00bfee-0cd7-4686-9dce-327bd9b9b96d.png)
This bug was fixed in commit d57fb4234363b2f27737450705b899c9a4306f55 by making `becho` only remove the last end sequence placed when applying an style, maintaining the ones that was inserted by a sub command. Now, the output fixed looks like:
![fix](https://user-images.githubusercontent.com/119078587/222768224-0547cceb-f2e2-421a-a9f4-618ca6f55d0f.png)
